pytorch_lightning/trainer/predict_loop.py

```diff
@@ -11,19 +11,47 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+from typing import Any, List, Optional
+
 import torch
 
-from pytorch_lightning.utilities.apply_func import apply_to_collection
+from pytorch_lightning.overrides.distributed import IndexBatchSampler
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
 from pytorch_lightning.utilities.warnings import WarningCache
 
 
 class PredictLoop(object):
 
+    write_intervals = ("step", "epoch")
+
     def __init__(self, trainer):
         self.trainer = trainer
         self.max_batches = None
         self.num_dataloaders = None
         self.warning_cache = WarningCache()
+        self.batch_indices: Optional[List[int]] = []
+        self._return_predictions: bool = not trainer.training_type_plugin.use_spawn
```

Comment:
No need for type annotation here, right? it will always be bool

```suggestion
        self._return_predictions = not trainer.training_type_plugin.use_spawn
```